content/kb/tutorials/databases/snowflake.md

```diff
@@ -60,107 +52,157 @@ Be sure to highlight or select **all** the queries (lines 1-10) before clicking
 
-1. Determine the version of the Snowflake Connector for Python you want to install.
-2. Determine the version of Python you want to use on Streamlit Community Cloud.
-3. To install the connector and the dependent libraries, select the [requirements file](https://github.com/snowflakedb/snowflake-connector-python/tree/main/tested_requirements) for that version of the connector and Python.
-4. Add the raw GitHub URL of the requirements file to your `requirements.txt` file and prepend `-r` to the line.
-   For example, if you want to install version `2.7.9` of the connector on Python 3.9, add the following line to your `requirements.txt` file:
+# Perform query.
+df = conn.query('SELECT * from mytable;', ttl=600)
 
-   ```text
-   -r https://raw.githubusercontent.com/snowflakedb/snowflake-connector-python/v2.7.9/tested_requirements/requirements_39.reqs
-   ```
+# Print results.
+for row in df.itertuples():
+    st.write(f"{row.NAME} has a :{row.PET}:")
+```
 
-5. On Streamlit Community Cloud, select the appropriate version of Python for your app by clicking "Advanced settings" before you deploy the app:
+See `st.experimental_connection` above? This handles secrets retrieval, setup, query caching and retries. In this case, we set `ttl=600` to ensure the query result is cached for no longer than 10 minutes. Watch out: If your database updates more frequently, you should adapt `ttl` or remove caching so viewers always see the latest data. Learn more in [Connecting to data](/library/advanced-features/connecting-to-data) and [Caching](/library/advanced-features/caching).
 
-<div style={{ maxWidth: '65%', marginBottom: '-1em', marginLeft: '6em', marginTop: '-2em' }}>
-    <Image src="/images/streamlit-community-cloud/advanced-settings.png" />
-</div>
+If everything worked out (and you used the example table we created above), your app should look like this:
 
-That's it! You're ready to use the Snowflake Connector for Python on Streamlit Community Cloud. ❄️🎈
+![Finished app screenshot](/images/databases/snowflake-app.png)
 
-<Tip>
+### Using a Snowpark Session
 
-As the Snowflake dependencies [requirements files](https://github.com/snowflakedb/snowflake-connector-python/tree/main/tested_requirements) (`.reqs`) contain the pinned version of the connector, there is **no need** add a separate entry for the connector to requirements.txt.
+The same [SnowparkConnection](/library/api-reference/connections/st.connections.snowparkconnection) used above also provides access to the [Snowpark Session](https://docs.snowflake.com/en/developer-guide/snowpark/reference/python/session.html) for DataFrame-style operations that run natively inside Snowflake. Using this approach, you can rewrite the app above as follows:
 
-</Tip>
+```python
+# streamlit_app.py
 
-## Write your Streamlit app
+import streamlit as st
+
+# Initialize connection and access underlying Snowpark session.
+session = st.experimental_connection('snowpark').session
```

Comment:
Hm, I think we probably want to direct people toward using `safe_session` here since I'd imagine that many users following this tutorial won't always know what assumptions they're able to make about thread safety / be expected to be able to guarantee access to the `session` object is thread safe.